containers/kubernetes/reference-content/exposing-services.mdx

```diff
@@ -0,0 +1,75 @@
+There are a number of different ways to expose your cluster to the internet. In this documentation, we explain and compare some of the different ways to do so, and provide information to help you get started.
+
+## Comparison of cluster exposure methods
+
+In Kubernetes, you generally need to use a [Service](/containers/kubernetes/concepts/#services) to expose an application in your cluster to the internet. A service groups together pods performing the same function (e.g. running the same application) and defines how to access them. 
+
+The most basic type of service is **ClusterIP**, but this only provides internal access, from within the cluster, to the defined pods. The **NodePort** and **LoadBalancer** services both provide external access. **Ingress** (which is not a service but an API object inside a cluster) combined with an explicitly-created **Ingress Controller** is another way to expose the cluster.
+
+See the table below for more information.
+
+
+| Method            | Description | Suitable for | Limitations |
+| ----------------- | ----------- | ------------ | ----------- |
+| **Cluster IP Service** | • Provides internal connectivity between cluster components. <br/>• Has a fixed IP address, from which it balances traffic between pods with matching labels. | Internal communication between different components within a cluster | Cannot be used to expose an application inside the cluster to the public internet. |
+| **Node Port Service**  | • Exposes a specific port on each node of a cluster. <br/>• Forwards external traffic received on that port to the right pods. | • Exposing single-node, low-traffic clusters to the internet for free <br/>• Testing. | • Not ideal for production or complex clusters <br/> • Single point of failure.<br/> • Not all port numbers can be opened.|
+| **Load Balancer Service** | • Creates a single, external Load Balancer with a public IP <br/> • External LB forwards all traffic to the corresponding LoadBalancer service within the cluster <br/> • LoadBalancer service then forwards traffic to the right pods. <br/> • Operates at the L4 level | • Exposing a service in the cluster to the internet. <br/>• Production envs (highly available). <br/>• Dealing with TCP traffic. | • Each service in the cluster needs its own Load Balancer (can become costly) |
+| **Ingress** | • A native resource inside the cluster (not a service) <br/>• Ingress Controller receives a single, external public IP, usually in front of a spun-up external HTTP Load Balancer <br/> • Uses a set of rules to forward web traffic (HTTP/S) to the correct service out of multiple services within the cluster. <br/> • Each service then sends the traffic to a suitable pod. <br/> • Operates at the L7 level.| • Clusters with many different services <br/> • Dealing with HTTP/S traffic | • Requires an Ingress Controller (not included by default, must be created). <br/> • Designed for HTTP/S traffic only (more complicated to configure for other protocols). | 
```

Comment:
@RoRoJ a good analogy is that the ingress controller is a L7 load balancer living inside the Kubernetes cluster and balancing traffic between other Kubernetes services (the original purpose is to route L7 traffic to the appropriate service --> a single entrypoint for multiple services in the cluster vs the 1:1 relationship between load balancer service and external load balancer).
The ingress controller is exposed to the internet using a load balancer service (there is an external load balancer deployed), and the external load balancer should also run as an L4 load balancer, HTTP/HTTPS is done in the ingress controller directly.

Reply:
@RoRoJ I just saw you nailed the benefits of ingress in the last section of the page, sorry for the lengthy comment.

Reply:
@RoRoJ I see you using the following in several places

> HTTP/S

Is this to imply HTTP or HTTPS? Asking because HTTPS doesn't require a / it's usually reserved for HTTP versions like HTTP/2.0